woocommerce/woocommerce-ios
# Coupons: Implement coupon deletion

Closes: #5773 


### Description

As the title suggests, this PR adds the implementation for deleting coupons. Changes include:
- Add a new feature flag for coupons M2: `couponEditAndDelete`.
- Add a new button on the coupon details screen's action sheet for deleting coupons if the feature flag for M2 is enabled.
- Add a confirm alert when coupon deletion is selected.
- Implement deleting a coupon.
- Show network progress on the navigation bar item and notice when the request succeeds or fails.

### Testing instructions

- Precondition: your test store should have coupons enabled (WP Admin > WC Settings > General > Enable Coupon) and has set up at least one coupon (WP Admin > Marketing > Coupons).
- With the `couponEditAndDelete` feature flag enabled, build and launch the app.
- Navigate to Menu > Coupons > select a coupon.
- On the coupon details screen, select the ellipsis (…) button, notice that there's a new destructive button Delete Coupon.
- Select the button, notice that there's an alert presented to confirm the deletion.
- Confirm deleting coupon, notice that a loading indicator is displayed on the trailing navigation bar item.
- If the request succeeds, notice that the coupon details screen is dismissed and a notice is presented on the coupon list saying "Coupon deleted". Otherwise, there should be a notice presented on the coupon details screen for the error.
- Please feel free to test with the feature flag off - the delete coupon button should not be present in that case.

### Screenshots


https://user-images.githubusercontent.com/5533851/159666674-684f45e9-cf51-4242-9cca-1b9740b2a076.mp4



---
